recipe/meta.yaml

```diff
@@ -774,8 +873,9 @@ outputs:
         - setuptools
         - setuptools-scm
       run:
-        - {{ pin_subpackage('pyarrow', exact=True) }}
-        - python
+        - {{ pin_subpackage('pyarrow-all', exact=True) }}
+        - {{ pin_compatible('numpy') }}
+        - {{ pin_compatible('python') }}
```

Comment:
@h-vetinari this is where I get confused on how conda gets the dependencies and solves the environment. I am not sure why but now that the `host` section is removed from `pyarrow-all` I had to add the `pin_compatible` to numpy and python otherwise it uses a different version than the one used to build. Example of the python error:
```
Could not solve for environment specs
The following packages are incompatible
├─ pyarrow-all 16.0.0.* ha770c72_0 is installable and it requires
│  └─ pyarrow 16.0.0 ha770c72_0, which requires
│     └─ pyarrow-core 16.0.0 py39he338278_0_cuda, which requires
│        └─ python_abi 3.9.* *_cp39, which can be installed;
└─ python 3.12.3.* hab00c5b_0_cpython is not installable because it requires
   └─ python_abi 3.12.* *_cp312, which conflicts with any installable versions previously reported.
```
I am not sure if this is the correct solution.

Reply:
Well, this clearly doesn't fix the issue:
```
Could not solve for environment specs
The following packages are incompatible
├─ numpy 1.22.4.* py39hc58783e_0 is requested and can be installed;
└─ pyarrow-all 16.0.0.* ha770c72_0 is not installable because it requires
   └─ pyarrow 16.0.0 ha770c72_0, which requires
      └─ pyarrow-core 16.0.0 py311h225a1a7_0_cuda, which requires
         └─ numpy >=1.23.5,<2.0a0 , which conflicts with any installable versions previously reported.
```

Reply:
Ah, OK, yeah, we should still use it, but only where we need the run-exports. E.g.
```
  host:
      # only necessary for run-exports
      - python
      - numpy
```